### DISCUSSION TITLE
Welcome to VuFind Discussions!

### DISCUSSION THREAD

**Discussion by demiankatz** [vufind-org/vufind#2379] [Announcements]
## Welcome!
  We’re trying out Discussions as a new place to conduct community conversations. For now, we still encourage [JIRA](https://vufind.org/jira) for bug reports/enhancement requests. Also, please remember that [additional channels](https://vufind.org/vufind/support.html) are available for support if you need it. Feel free to experiment with this new tool, and let us know what you think!

**Comment by Eng-Luke** [comment]
Hello I am still having challenges Installing my vufind. And after following all the necessary steps I still get an error. It would be wonderful if you have like live support with someone who can see your problem and somehow tries to fix it

**Reply by Eng-Luke** [reply]
Something like this 
![Screenshot_20230626_104148_Chrome](https://github.com/vufind-org/vufind/assets/129087829/6bed100f-1938-4664-844b-43631b3c09dc)
My link being: http://167.172.185.75/vufind/

**Comment by demiankatz** [comment]
@Eng-Luke, see the [support page](https://vufind.org/vufind/support.html) for other support options. You may be able to have real time discussions on Slack if that is your preferred method, though keep in mind we can't guarantee anything as this is a volunteer-driven project. If you want guaranteed support, several paid services are also listed on that page. 

Regarding the error you are seeing, please consult the [troubleshooting page](https://vufind.org/wiki/development:troubleshooting) on how to turn on development mode. This will reveal a more detailed error message that will make troubleshooting easier. If it doesn't help you solve the problem yourself, you can share more details here or in one of our other support channels to get further help.

**Comment by Eng-Luke** [comment]
Noted thank you

**Reply by demiankatz** [reply]
Of course, always happy to help. Just want to set realistic expectations since, especially in summer when there are a lot of vacations, the number of available volunteers can be a bit limited at any given moment. :-)

**Comment by Eng-Luke** [comment]
Awesome, thank you so much! So can we discuss further on and see how best
you may assist me and how best I can express my gratitude?

**Reply by demiankatz** [reply]
Of course, feel free to let me know what you need, and I'll do my best to connect you with somebody who can help. My own availability is somewhat limited because I'm focused on managing and maintaining the project, which already takes up a lot of my available time... but I always try to answer questions as promptly as I can.

**Comment by Eng-Luke** [comment]
So I followed the Instructions perfectly and I have tried deleting and
reinstalling again but I keep getting these errors. I was thinking of
creating a droplet for you with shared details then we try to go through it
together otherwise I would be willing to fly abroad to learn more on
vufind. Let me know if you are okay with that and I can share my details
with you

**Reply by demiankatz** [reply]
@Eng-Luke, can you share the exact errors you are seeing after [turning on development mode](https://vufind.org/wiki/development:troubleshooting#turn_on_development_mode)? If I know more details about your problem, I can most likely help you to troubleshoot.

If you are interested in meeting other VuFind users in person, you might consider attending [WOLFcon](https://openlibraryfoundation.org/about/wolfcon/upcoming/), which is the home of the annual VuFind Summit. I will be there in person, and we could probably find some time during the conference to work together. However, I suspect it will be possible to figure this out without putting you to so much inconvenience. (Also note that you can virtually attend WOLFcon for free, so if you'd like to see the VuFind content without having to travel, that is an option too, and you might find some of it interesting).

**Comment by Eng-Luke** [comment]
Kindly we request a Video Conference with some of my colleagues so that we
can learn on Vufind for at most an hour. We are willing to pay for your
time in the PayPal method. If the option is possible

**Reply by demiankatz** [reply]
The next [VuFind Community Call](https://vufind.org/wiki/community_call) is scheduled for July 11 at 9am Eastern time. That monthly call is a good opportunity to ask questions of the community and get help. If you can wait that long, that might be a good opportunity to discuss this in real time (and I don't mind staying a little late after the call if necessary).

If you want a much deeper introduction to VuFind, you might also be interested in the training program being offered in August by @livedvd from HealthNet Nepal. There is a [registration form](https://docs.google.com/forms/d/e/1FAIpQLSfmy-RflNVd5vd--IeGtaafotIk9RFvdBJ5opF0oUy5U9pJjQ/viewform) with more information.

If you need to talk before those events, please email me (demian.katz@villanova.edu) with your schedule/availability and I will see if we can find a time to chat. It is easier for me to offer support asynchronously, however, because my calendar tends to be very full. In any case, there is no need to pay me -- if you are happy with the software and community support, there will be a project fundraiser coming up later this summer, and you're certainly welcome to contribute to that if you wish.

**Comment by iftekhar92** [comment]
@demiankatz 
I am setup [vufind ](https://vufind.org/vufind/) on my ubuntu 20.04 VPS server. I followed Installation [Ubuntu ](https://vufind.org/wiki/installation:ubuntu) documentation. All steps have gone well. But, when I tried to hit Installation [link ](https://discoveryconnect.in/Install/Home) then I am getting error like:
![image](https://github.com/user-attachments/assets/1a514e8d-09ac-42dc-a4dc-38c462515d57)


**Here is my configurations:**

![image](https://github.com/user-attachments/assets/3782770d-4030-4829-b094-84581f2c4c42)

**cat /var/www/discoveryconnect.in/public_html/local/httpd-vufind.conf**
![image](https://github.com/user-attachments/assets/e890e1ad-5826-427c-b4b6-379b24e63e65)

Can someone please answer I will be grateful.

**Reply by demiankatz** [reply]
@iftekhar92, what version of VuFind are you installing? Ubuntu 20.04 is very old (it reaches "end of life" this April), and it includes PHP 7.4. Newer versions of VuFind require newer versions of PHP. You would be better off installing Ubuntu 24.04 if at all possible, if you want to run an up-to-date VuFind version and keep your system updated and secure for more than three months.

The specific error you are seeing indicates that a constant expected in the mysqli PHP extension is undefined -- this could mean that the mysqli extension is not installed, but it might also be a side effect of running an outdated PHP version.

I hope this is some help -- please let me know if you still have questions or need more help!

**Reply by iftekhar92** [reply]
@demiankatz Thank you so much for you quick response.
I have upgraded my server from Ubuntu 24.04. Now, I am able to see running [app ](https://discoveryconnect.in) by running the below commands like:

```
1. wget https://github.com/vufind-org/vufind/releases/download/v10.1/vufind_10.1.deb
2. sudo dpkg -i vufind_10.1.deb
3. sudo apt-get install -f (Optional, if step 2 gets failed).
```

**Reply by demiankatz** [reply]
Great, I'm glad I could help! Please let me know if you need further support. (I see your instance is currently displaying an error message, but perhaps this is because you are currently working on it; hopefully the [troubleshooting](https://vufind.org/wiki/development:troubleshooting) page will be useful if you need it). :-)

**Comment by iftekhar92** [comment]
Hi Team,

I am facing a challenge while solving the DB connection issue Install/Home
<https://discoveryconnect.in/Install/FixDatabase>.

   - I am using MySQL Mariadb mysql  Ver 15.1 Distrib 10.6.18-MariaDB, for
   debian-linux-gnu (x86_64) using  EditLine wrapper
   - I followed the steps as mentioned in the installation page.
   - After filling for when I submit the page then it's showing error
   message  Connection error
   - As per my understanding, it seems to be a db connection error. But,
   when I run a dump query (provided by itself that is attached) on VPS server
   then it is running well.

I need support, Please help me and let me know if I need more info from my
side.


Thanks,
Iftekhar Ahmad

**Reply by demiankatz** [reply]
@iftekhar92, I don't see the attachment that you refer to. In order for the web-based database installation to work, the root database account needs to be set up with an authentication method that accepts a username and password. In some distributions, the default setup uses socket authentication, which complicates connections via Apache. This configuration can vary a lot from distribution to distribution; in Fedora Linux, MariaDB works fine "out of the box." In Ubuntu Linux, MySQL requires some reconfiguration (as described on the [installation instructions page](https://vufind.org/wiki/installation:ubuntu#example_2using_mariadb_instead_of_mysql)) to work properly. It seems that whatever flavor of Debian you are using has a MariaDB configuration that does not work by default. There are a few options here:

1.) You can examine the database configuration and account settings and try to figure out why connections are failing. I have very limited experience with MariaDB (I use MySQL in my environment), and no one in the community seems to have shared a solution to this particular problem, so I unfortunately can't provide much specific guidance... but looking at the MySQL notes and the linked MariaDB documentation on the page I mentioned above might help.

2.) If you can't get MariaDB to work, you could try MySQL or PostgreSQL and see if it's easier to get those working.

3.) If you're able to connect to MariaDB from the command line, you may be able to bypass the web-based installation by using the "skip" button, which gives you SQL you can copy and paste into a file and run through the command-line tool to generate the necessary database tables. As long as the new account that you create for the purposes of VuFind connectivity has appropriate settings, then it should work without requiring you to manipulate your root account. However, if MariaDB is set up with defaults that are incompatible with PHP/Apache, the new account might end up having the same problem that the root account currently has.

I hope this is some help; I'm sorry that I don't have a more specific solution. Unfortunately, with the wide array of Linux distributions and databases, I am not able to test every possible combination, and sometimes problems like this arise.

**Comment by iftekhar92** [comment]
Thank You so much for your kind of support.

I have run the query manually in MariaDB and then also updated the DB
details in config.ini. Now, I see the DB issue is resolved.

**Reply by demiankatz** [reply]
Excellent, I'm glad this worked for you, and I'm sorry that the more convenient automatic method didn't work for you.

**Comment by ccorsani** [comment]
Hi, we try to use an email delivery service where smtp user is not in the form "user@domain" but only "user". How can we setup such smtp?

**Reply by demiankatz** [reply]
The `username` setting in the `[Mail]` section of config.ini determines the user used for SMTP authentication. I don't believe that there are any restrictions on the format of this value. Is it not working as expected?

**Reply by ccorsani** [reply]
the answer:

this is the answer: 2026-02-03T10:13:05+01:00 CRIT (2): Laminas\Mail\Exception\InvalidArgumentException : The input is not a valid email address. Use the basic format local-part@hostname

**Reply by demiankatz** [reply]
Is it possible that a different setting is causing that error (e.g. in invalid from/to address), or have you confirmed that it is definitely related to validation of the username parameter?

In any case, which version of VuFind are you using? We no longer use the Laminas\Mail library, as it has been deprecated, so this suggests that you are on an older version. It is possible that upgrading to VuFind 11, which replaces Laminas\Mail with Symfony\Mailer, will offer different possibilities.

**Reply by ccorsani** [reply]
This is our [MAIL] conf section:

email = "noreply@mydomain"
host  = delivery.mailservice.com
port  = 587
secure  = tls
username  = “smtp_user”
password  = “smtp_password“

**Reply by ccorsani** [reply]
Found this: https://github.com/glpi-project/glpi/issues/19037

seems a bug of Lamina, solved at the end of 2025, isn't it?

**Reply by demiankatz** [reply]
laminas-mail was abandoned in 2023 and has not had any updates since that time; that is the reason we were forced to switch to symfony-mailer in VuFind 11. If there is a Laminas bug responsible for the problem, the best solution would be to upgrade to VuFind 11 so you can take advantage of the symfony-mailer switch.

If possible, I would suggest at least setting up an instance of VuFind 11 in a test environment so you can identify whether that resolves the problem. If it does, then an upgrade may be your best solution. If it does not, at least we know that we have some other layer of troubleshooting that we will have to work through.
